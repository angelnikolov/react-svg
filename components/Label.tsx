import React from 'react';
import * as styledComponents from 'styled-components';

export interface ILabelProps {
  text: string;
  x: number;
  y: number;
  size?: number;
}

const StyledLabel = styledComponents.default.text`
  fill: #888;
  font-size:${(props: ILabelProps) => props.size ? props.size : 18}px;
`;

const Label: React.StatelessComponent<ILabelProps> = props => (
  <StyledLabel {...props}>{props.text}</StyledLabel>
);

export default Label;
