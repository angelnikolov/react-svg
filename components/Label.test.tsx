// Label.test.js
import 'jest-styled-components';

import { shallow } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';

import Label from './Label';

const enzyme = require("enzyme");
const Adapter = require("enzyme-adapter-react-16");
enzyme.configure({ adapter: new Adapter() });
test("Label should render", () => {
  const component = renderer.create(
    <Label text="Test" x={15} y={15} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

});

test("shallow test", ()=>{
  const label = shallow(<Label text="Test" x={15} y={15} />);
  expect(label.text()).toBe('<styled.text />')
})