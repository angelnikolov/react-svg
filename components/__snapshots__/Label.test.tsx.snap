// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label should render 1`] = `
.c0 {
  fill: #888;
  font-size: 18px;
}

<text
  className="c0"
  x={15}
  y={15}
>
  Test
</text>
`;
